tests/api2/test_440_snmp.py

```diff
@@ -63,65 +132,254 @@ def validate_snmp_get_sysname_uses_same_ip(hostip):
     assert get_dst.endswith(".161")
 
 
-def test_01_Configure_SNMP():
-    results = PUT('/snmp/', {
+def user_list_users(snmp_config):
+    """Run an snmpwalk as a SNMP v3 user"""
+
+    add_cmd = None
+    if snmp_config['v3_privproto']:
+        authpriv_setting = 'authPriv'
+        add_cmd = f"-x {snmp_config['v3_privproto']} -X \"{snmp_config['v3_privpassphrase']}\" "
+    else:
+        authpriv_setting = 'authNoPriv'
+
+    cmd = f"snmpwalk -v3 -u  {snmp_config['v3_username']} -l {authpriv_setting} "
+    cmd += f"-a {snmp_config['v3_authtype']} -A {snmp_config['v3_password']} "
+    if add_cmd:
+        cmd += add_cmd
+    cmd += "localhost iso.3.6.1.6.3.15.1.2.2.1.3"
+
+    # This call will timeout if SNMP is not running
+    res = ssh(cmd)
+    return [x.split()[-1].strip('\"') for x in res.splitlines()]
+
+
+@pytest.mark.dependency(name='SNMP_CONFIGURED', scope="module")
```

Comment:
What's the benefit of using this and not just fixtures?

Reply:
'This' being the dependency mark or?
I'm not sure I understand your comment.  Could you give an example?

Reply:
Yes, "this" being the dependency mark. Why do we need test dependencies?

If we want to skip executing a test if something fails to initialize, fixtures will do this perfectly.

Reply:
OK, I see.  Yes, that's a nice improvement!  Thanks